src/Libraries/Microsoft.Extensions.Telemetry/Logging/Logger.cs

```diff
@@ -3,177 +3,70 @@
-                typeof(LogLevel),
-                typeof(EventId),
-                typeof(string),
-                typeof(object),
-                typeof(Exception),
-                typeof(IReadOnlyList<KeyValuePair<string, object?>>)
-            },
-            null)!;
-
-        var val = new[]
-        {
-            Expression.Parameter(typeof(IExternalScopeProvider)),
-            Expression.Parameter(typeof(DateTime)),
-            Expression.Parameter(typeof(string)),
-            Expression.Parameter(typeof(LogLevel)),
-            Expression.Parameter(typeof(EventId)),
-            Expression.Parameter(typeof(string)),
-            Expression.Parameter(typeof(object)),
-            Expression.Parameter(typeof(Exception)),
-            Expression.Parameter(typeof(IReadOnlyList<KeyValuePair<string, object?>>))
-        };
-
-        var lambdaLogRecord = Expression.Lambda<Func<IExternalScopeProvider?, DateTime, string, LogLevel,
-            EventId, string?, object?, Exception?, IReadOnlyList<KeyValuePair<string, object?>>?,
-            LogRecord>>(Expression.New(logRecordConstructor, val), val);
-
-        return lambdaLogRecord.Compile();
-    }
+    // TODO From Noah: For stack traces I'd recommend using new StackTrace(Exception, fNeedFileInfo:false)
+    // or you might incur major perf penalties and contention if a service logs a bunch of stack traces when a PDB is present on disk
 
-    internal static TimeProvider TimeProvider => TimeProvider.System;
+    public Logger(ILogger nextLogger, ILogEnricher[] enrichers, bool includeStackTraces, int maxStackTraceLength, IRedactorProvider? redactorProvider)
+    {
+        _nextLogger = nextLogger;
+        _enrichers = enrichers;
+        _includeStackTraces = includeStackTraces;
+        _maxStackTraceLength = maxStackTraceLength;
+        _redactorProvider = redactorProvider;
+    }
 
-    internal Logger(string categoryName, LoggerProvider provider)
+    public IDisposable? BeginScope<TState>(TState state)
+        where TState : notnull
     {
-        _categoryName = categoryName;
-        _provider = provider;
+        return _nextLogger.BeginScope(state);
     }
 
-    internal IExternalScopeProvider? ScopeProvider { get; set; }
+    public bool IsEnabled(LogLevel logLevel)
+    {
+        // TODO: is this the best we can do here? Should/could the enabled state be tracked in this instance so we can avoid the interface call?
+        return _nextLogger.IsEnabled(logLevel);
+    }
 
     public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception? exception, Func<TState, Exception?, string> formatter)
     {
-        if (!IsEnabled(logLevel))
-        {
-            return;
-        }
-
-        LogMethodHelper propertyBag;
-        LogMethodHelper? rentedHelper = null;
-
         try
         {
-            if (state is LogMethodHelper helper && _provider.CanUsePropertyBagPool)
+            if (typeof(TState) == typeof(LoggerMessageState))
```

Comment:
Can we do a pattern matching here? The structure will look much cleaner

Reply:
I'm using a specific layout here to allow the JIT to eliminate the code paths and conditional checks that aren't needed at runtime when the TState is known statically.